src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -253,6 +257,59 @@ private void Initialize(CancellationToken cancellationToken = default)
         }
     }
 
+    internal static class StreamTimeoutExtensions
+    {
+        internal static async Task<int> ReadAsync(this Stream stream, Memory<byte> buffer, TimeSpan readTimeout, CancellationToken cancellationToken)
```

Comment:
Why didn't you use perReadTimeout here?

Reply:
There is only one read here so it doesn't make sense to call it per read timeout.

Reply:
Maybe you could just always use `readTimeout`

Reply:
I think it's clearer that the read timeout applies to every read in the copyToAsync with the current name rather than the copy as a whole.